### ISSUE TITLE
Backup-DbaDatabase - erorr with -Verify switch

### ISSUE THREAD

**Issue by Zhunya** [sqlcollaborative/dbatools#2717]
## Bug Report

`Test-DbaSqlPath` throws error on testing path, wrong variable/value passed to it from `Backup-DbaDatabase` at `Backup-DbaDatabase`, line 481

### General Troubleshooting steps

- [x] Verified running latest release of dbatools?
- [x] Verified errors are not related to permissions?
- [x] Can duplicate in new/fresh PowerShell.exe session?

### Version Information
 - Operating System (Name|Version): Windows 10, 1709
 - PowerShell Version: 5.1
 - SQL Server (Edition|Version): SQL Server 2008 R2 +

## Steps to Reproduce

Run `Backup-DbaDatabase` with `-Verify` switch
```powershell
writeErrorStream      : True
PSMessageDetails      :
Exception             : System.Exception: SqlInstance SEDNA\INST14 cannot access @{SqlInstance=sedna\inst14; DatabaseName=v8_2011_audit_16fh; BackupComplete=True; BackupFilesCount=1; BackupFile=v8_2011_audit_16fh_201711221556.bak; BackupFolder=\\SRV-BACKUP\SQLBackup\Audits\; BackupPath=\\SRV-BACKUP\SQLBackup\Audits\\v8_2011_audit_16fh_201711221556.bak; Script=BACKUP DATABASE [v8_2011_audit_16fh] TO  DISK = N'\\SRV-BACKUP\SQLBackup\Audits\\v8_2011_audit_16fh_201711221556.bak' WITH NOFORMAT, NOINIT, NOSKIP, REWIND, NOUNLOAD, COMPRESSION,  STATS = 1, CHECKSUM; Notes=; FullName=\\SRV-BACKUP\SQLBackup\Audits\\v8_2011_audit_16fh_201711221556.bak; FileList=System.Object[]; SoftwareVersionMajor=12;Type=Full; FirstLsn=509026000000014600001; DatabaseBackupLsn=508669000003095200001; CheckPointLsn=509026000000014600001; LastLsn=509026000000014900001; BackupSetId=19381; LastRecoveryForkGUID=978354c3-8d7c-4a15-9a66-c55d4a4e903a}\ ---> System.Exception: SqlInstance SEDNA\INST14 cannot access @{SqlInstance=sedna\inst14; DatabaseName=v8_VelesCorporate2011_audit_16fh; BackupComplete=True; BackupFilesCount=1; BackupFile=v8_2011_audit_16fh_201711221556.bak; BackupFolder=\\SRV-BACKUP\SQLBackup\Audits\; BackupPath=\\SRV-BACKUP\SQLBackup\Audits\\v8_2011_audit_16fh_201711221556.bak; Script=BACKUP DATABASE [v8_2011_audit_16fh] TO  DISK = N'\\SRV-BACKUP\SQLBackup\Audits\\v8_2011_audit_16fh_201711221556.bak' WITH NOFORMAT, NOINIT, NOSKIP, REWIND, NOUNLOAD, COMPRESSION,  STATS = 1, CHECKSUM; Notes=; FullName=\\SRV-BACKUP\SQLBackup\Audits\\v8_2011_audit_16fh_201711221556.bak; FileList=System.Object[]; SoftwareVersionMajor=12; Type=Full; FirstLsn=509026000000014600001; DatabaseBackupLsn=508669000003095200001; CheckPointLsn=509026000000014600001; LastLsn=509026000000014900001; BackupSetId=19381; LastRecoveryForkGUID=978354c3-8d7c-4a15-9a66-c55d4a4e903a}\
                           --- End of inner exception stack trace ---
TargetObject          :
CategoryInfo          : NotSpecified: (:) [Write-Error], Exception
FullyQualifiedErrorId : dbatools_Backup-DbaDatabase,Stop-Function
ErrorDetails          : SqlInstance SEDNA\INST14 cannot access @{SqlInstance=sedna\inst14; DatabaseName=v8_2011_audit_16fh; BackupComplete=True; BackupFilesCount=1; BackupFile=v8_2011_audit_16fh_201711221556.bak; BackupFolder=\\SRV-BACKUP\SQLBackup\Audits\; BackupPath=\\SRV-BACKUP\SQLBackup\Audits\\v8_2011_audit_16fh_201711221556.bak; Script=BACKUP DATABASE [v8_2011_audit_16fh] TO  DISK = N'\\SRV-BACKUP\SQLBackup\Audits\\v8_2011_audit_16fh_201711221556.bak' WITH NOFORMAT, NOINIT, NOSKIP, REWIND, NOUNLOAD, COMPRESSION,STATS = 1, CHECKSUM; Notes=; FullName=\\SRV-BACKUP\SQLBackup\Audits\\v8_2011_audit_16fh_201711221556.bak; FileList=System.Object[]; SoftwareVersionMajor=12; Type=Full; FirstLsn=509026000000014600001; DatabaseBackupLsn=508669000003095200001; CheckPointLsn=509026000000014600001; LastLsn=509026000000014900001; BackupSetId=19381; LastRecoveryForkGUID=978354c3-8d7c-4a15-9a66-c55d4a4e903a}\
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Stop-Function, <No file>: line 198
                        at Backup-DbaDatabase<Process>, <No file>: line 481
                        at <ScriptBlock>, <No file>: line 1
```

**Comment by Stuart-Moore** [comment]
Fixed that in the restoreLSN branch. #2712 

Needed some tweaks in files I'd already modified so it saved a nasty merge later

**Comment by Stuart-Moore** [comment]
Fixed in #2712 and confirmed by @Zhunya
